test/framework/src/main/java/org/elasticsearch/test/ESTokenStreamTestCase.java

```diff
@@ -29,7 +28,7 @@
  * Basic test case for token streams. the assertion methods in this class will
  * run basic checks to enforce correct behavior of the token streams.
  */
-public abstract class ESTokenStreamTestCase extends BaseTokenStreamTestCase {
+public abstract class ESTokenStreamTestCase extends ESTestCase {
```

Comment:
Is this and all the other refactoring only to get the `assertWarnings` action?

I would much rather us just add `assertWarnings` to `ESTokenStreamTestCase`. Its pretty simple. See: https://github.com/elastic/elasticsearch/pull/113143/files

I can do that in another PR really quick.

Reply:
Yes, that was the motivation, plus the realization that `BaseTokenStreamTestCase` doesn't give us much except a bunch of static helper methods but makes any test derived from that different from our usual ESTestCase, so I thought switching that over wouldn't hurt.

Reply:
No strong feelings either way.

Reply:
https://github.com/elastic/elasticsearch/pull/113619

Reply:
My PR is now merged. So, maybe we can update this PR to clean it up a bit?